### ISSUE TITLE
🐞 crew install (package) fails when installing some programs

### ISSUE THREAD

**Issue by hexisXz** [chromebrew/chromebrew#10388]
### bug 
For example i was trying to install crew_launcher and it failed with this error 

`/usr/local/bin/crew:710:in `chdir': No such file or directory @ dir_chdir0 - /home/chronos/user/.local/bin (Errno::ENOENT)
        from /usr/local/bin/crew:710:in `pre_install'
        from /usr/local/bin/crew:1283:in `install'
        from /usr/local/bin/crew:1217:in `block in resolve_dependencies'
        from /usr/local/bin/crew:1214:in `each'
        from /usr/local/bin/crew:1214:in `resolve_dependencies'
        from /usr/local/bin/crew:1107:in `resolve_dependencies_and_install'
        from /usr/local/bin/crew:1695:in `block in install_command'
        from /usr/local/bin/crew:1680:in `each'
        from /usr/local/bin/crew:1680:in `install_command'
        from /usr/local/bin/crew:1826:in <main>'
crew_launcher isn't the only package that fails to install. If i remember correctly i think flatpak does as well.

**Reproduce**
crew install (package)

i use bash and shell in crosh not VT-2 or whatever the tty is 

**Expected behavior**
crew was supposed to install the program but the program doesn't install and crew exits with errors 

***
- Architecture: `aarch64` (`armv7l`)
- Processor vendor: `ARM`
- User space: `64-bit`
- Chromebrew Kernel version: `5.15`
- Chromebrew Running in Container: `false`

- Chromebrew version: `1.51.4`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib`

- Last update in local repository: 386f42ff6 `Fix for require issues with highline and glibc 2.37 refactor (#10363) (3 hours ago)`

- OS variant: `Chrome OS`
- OS version: `corsola-release/R127-15917.78.0`
- OS channel: `stable-channel`
***

**Additional context**
OH and after running ANY other command that was installed with crew after getting the error message from installing a program i now get this error when trying to run any command 
`sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory`

the programs still work but those error will display first then the program will run

# EDIT 
actually EVERYTHING breaks meaning I cant install any program with crew and now crew is broken so I can't install any program and I get the same error message everytime

**Comment by uberhacker** [comment]
Sounds like you have a corrupted install.  If you don't have anything you want to save in `/usr/local`, I would suggest the following:
```
rm -rf /usr/local
bash <(curl -L git.io/vddgY) && . ~/.bashrc
```

**Comment by hexisXz** [comment]
> Sounds like you have a corrupted install. If you don't have anything you want to save in `/usr/local`, I would suggest the following:
> 
> ```
> rm -rf /usr/local
> bash <(curl -L git.io/vddgY) && . ~/.bashrc
> ```

it wont let me remove the directory and says   `rm: cannot remove '/usr/local': Read-only file system`

**Comment by hexisXz** [comment]
> Sounds like you have a corrupted install. If you don't have anything you want to save in `/usr/local`, I would suggest the following:
> 
> ```
> rm -rf /usr/local
> bash <(curl -L git.io/vddgY) && . ~/.bashrc
> ```

well I guess it worked but now I cant reinstall crew and I get this error `curl: (77) error setting certificate file: /usr/local/etc/ssl/certs/ca-certificates.crt
bash: /usr/local/etc/profile: No such file or directory`

**Comment by uberhacker** [comment]
You can ignore that error.  If you cannot install crew try a powerwash assuming there is nothing you want to save.  https://www.youtube.com/watch?v=3nZP7tgi6ME

**Comment by hexisXz** [comment]
> You can ignore that error. If you cannot install crew try a powerwash assuming there is nothing you want to save. https://www.youtube.com/watch?v=3nZP7tgi6ME

dang.. this would be the third time today I powerwashed chromeos. well I will do it again and tell you if it works. but the install bug is still a problem because it doesn't just happen on this laptop so I just can't install apps like flatpak or the one launcher program or it will break my system again.. BUT I did realize something, after i installed crew i ran sudo dev_install but i am not sure if this would be the cause of the problem.

**Comment by uberhacker** [comment]
Yes, you don't want to run anything with sudo because it changes permissions and that might be your issue.

**Comment by hexisXz** [comment]
> Yes, you don't want to run anything with sudo because it changes permissions and that might be your issue.

well dev_install requires sudo so idk what I did wrong but I did try crew without running the other command and yeah something is broken in crew that's just not allowing my to install some packages

**Comment by uberhacker** [comment]
@satmandu?

**Comment by satmandu** [comment]
Wait, Chromebrew is being installed on top of `dev_install`?

I don't think that's going to work...

**Comment by hexisXz** [comment]
> Wait, Chromebrew is being installed on top of `dev_install`?
> 
> I don't think that's going to work...

Yeah I figured that out but like I said earlier (at least I think I sad) I tried installing chromebrew without installing or ever running the `dev_install` command and chromebrew still breaks when installing some packages like it corrupts the ENTIRE chromebrew install when installing packages like gimp xfce terminal flatpak and others

**Comment by satmandu** [comment]
Hmm. I wonder if this is an aarch64 issue. We don't actually have an aarch64 bootstrap at the moment, so all of our packages are actually armv7l, even when running on an aarch64 kernel.

**Comment by hexisXz** [comment]
> Hmm. I wonder if this is an aarch64 issue. We don't actually have an aarch64 bootstrap at the moment, so all of our packages are actually armv7l, even when running on an aarch64 kernel.

**Comment by hexisXz** [comment]
> Hmm. I wonder if this is an aarch64 issue. We don't actually have an aarch64 bootstrap at the moment, so all of our packages are actually armv7l, even when running on an aarch64 kernel.

dang so this is probably never going to be fixed?

**Comment by hexisXz** [comment]
> Hmm. I wonder if this is an aarch64 issue. We don't actually have an aarch64 bootstrap at the moment, so all of our packages are actually armv7l, even when running on an aarch64 kernel.

also flatpak.rb has 

`
class Flatpak < Meson
  description 'Flatpak is a system for building, distributing, and running sandboxed desktop applications on Linux.'
  homepage 'https://flatpak.org'
  version '1.15.6'
  license 'LGPL-2.1+'
  compatibility 'x86_64 aarch64 armv7l'
  source_url 'https://github.com/flatpak/flatpak.git'
  git_hashtag version
  binary_compression 'tar.zst'

  binary_sha256({
    aarch64: '313ff8ecaf84a2a42d24bde5999d6ec455dda6694d734973cb4c226b510241d9',
     armv7l: '313ff8ecaf84a2a42d24bde5999d6ec455dda6694d734973cb4c226b510241d9',
     x86_64: '205159d142b8fb7c9a65ac75309d4391889af4afc2dbd56e79da692b18c93e32'
  })`

so flatpak has aarch64 support but if this error can't be fixed then those packages might have to be changed to not be available on aarch64 because they will just break the system and  users will likely have to do a powerwash.

**Comment by satmandu** [comment]
> > Hmm. I wonder if this is an aarch64 issue. We don't actually have an aarch64 bootstrap at the moment, so all of our packages are actually armv7l, even when running on an aarch64 kernel.
> 
> also flatpak.rb has 
> 
> 
> ```
> class Flatpak < Meson
>   description 'Flatpak is a system for building, distributing, and running sandboxed desktop applications on Linux.'
>   homepage 'https://flatpak.org'
>   version '1.15.6'
>   license 'LGPL-2.1+'
>   compatibility 'x86_64 aarch64 armv7l'
>   source_url 'https://github.com/flatpak/flatpak.git'
>   git_hashtag version
>   binary_compression 'tar.zst'
> 
>   binary_sha256({
>     aarch64: '313ff8ecaf84a2a42d24bde5999d6ec455dda6694d734973cb4c226b510241d9',
>      armv7l: '313ff8ecaf84a2a42d24bde5999d6ec455dda6694d734973cb4c226b510241d9',
>      x86_64: '205159d142b8fb7c9a65ac75309d4391889af4afc2dbd56e79da692b18c93e32'
>   })
> ```
> 
> so flatpak has aarch64 support but if this error can't be fixed then those packages might have to be changed to not be available on aarch64 because they will just break the system and  users will likely have to do a powerwash.

Ses how the hashes are the same for aarch64 and armv7l? That's because they're currently using the same binaries.

We would be happy for someone to help bootstrap an aarch64 install... Just getting a glibc built would be a good start...
